app/models/document_type/body_field.rb

```diff
@@ -17,21 +17,23 @@ def updater_params(_edition, params)
     { contents: { body: params[:body] } }
   end
 
-  def pre_preview_issues(edition, revision)
+  def pre_update_issues(edition, params)
     issues = Requirements::CheckerIssues.new
 
-    unless GovspeakDocument.new(revision.contents[id], edition).valid?
+    unless GovspeakDocument.new(params[:contents][:body], edition).valid?
       issues.create(id, :invalid_govspeak)
     end
 
     issues
   end
 
-  alias_method :pre_update_issues, :pre_preview_issues
+  def pre_preview_issues(edition)
+    pre_update_issues(edition, contents: edition.contents.symbolize_keys)
```

Comment:
The introduction of contents here feels a bit weird, not sure if it's a smell - the symbolize keys makes me feel more like it's a smell as it feels like we're contorting attributes.

I did wonder if we might instead want the params for body to be just `params[:body]` and that this might feel a little less weird if it was `pre_update_issues(edition, body: edition.contents['body"])` - I figure that might either slot in quite nicely or create a brutal challenge of different places expecting different things.

Another thing I noticed in this class is that is that we seem to mix whether we use the id value or a body symbol. I'd imagine that for nearly all cases we could use the id as whenever we're using an ActionController::Parameters we have a hash with indifferent access.

Reply:
I agree, it does seem a little contrived. I can add a little explanation about the form of the params: for content checkers, the params that are passed in are the ones generated by the `updater_params` methods, since for `title_and_base_path` we need the `base_path` and don't want to be generating it twice. This seems reasonable, since the name of the method suggests we're going to be checking the things we're trying to use for the update. Doing it this way means the params have to be of the form `contents: { body: ... }`.

The pain point I had when implementing this is that `edition.contents` is not a has with indifferent access (or frankly, just a has with symbols). Switching everything to use string keys to accommodate this seems like a step backwards, so I'd prefer to try and address the stringiness of the JSON columns, if possible. Unfortunately, there doesn't seem to be an easy way of doing this without overriding methods :-(.